waiter/src/waiter/service.clj

```diff
@@ -262,7 +262,7 @@
                 instance)
               (if (and instance (not= instance :no-matching-instance-found))
                 ; instance is a deployment error if it (1) does not have an :id tag, (2) is not nil, and (3) does not equal :no-matching-instance-found
-                (ex-info (str "Deployment error: " (utils/message instance)) {:service-id service-id :status http-503-service-unavailable})
+                (ex-info (str "Deployment error: " (utils/message instance :default instance)) {:service-id service-id :status http-503-service-unavailable})
```

Comment:
Why is this necessary?

Reply:
The instance can either be an instance or a deployment error message. (utils/message) function deserializes the instance into a proper deployment error message, but in this situation instance can be a k8s api error message. So if instance message is not found in deployment messages map, then we default to instance message itself.

Reply:
I would use this then:
```suggestion
                (ex-info (str "Deployment error: " (or (utils/message instance) instance)) {:service-id service-id :status http-503-service-unavailable})
```